desc/compute/_geometry.py

```diff
@@ -355,29 +421,42 @@ def _a_major_over_a_minor(params, transforms, profiles, data, **kwargs):
         )
         / max_rho
     )
-    # surface area
+    # surface area. Need to recompute this since the grids used for A and P are
+    # usually different
     A = surface_integrals(
         transforms["grid"],
         jnp.abs(data["sqrt(g)"] / data["R"]),
         surface_label="zeta",
         expand_out=False,
     )
-    # derived from Ramanujan approximation for the perimeter of an ellipse
-    a = (  # semi-major radius
-        jnp.sqrt(3)
-        * (
-            jnp.sqrt(8 * jnp.pi * A + P**2)
-            + jnp.sqrt(
-                jnp.abs(
-                    2 * jnp.sqrt(3) * P * jnp.sqrt(8 * jnp.pi * A + P**2)
-                    - 40 * jnp.pi * A
-                    + 4 * P**2
-                )
-            )
-        )
-        + 3 * P
-    ) / (12 * jnp.pi)
-    b = A / (jnp.pi * a)  # semi-minor radius
+    a, b = _ellipse_radii(P, A)
+    data["a_major/a_minor"] = transforms["grid"].expand(a / b, surface_label="zeta")
+    return data
+
+
+@register_compute_fun(
+    name="a_major/a_minor",
+    label="a_{\\mathrm{major}} / a_{\\mathrm{minor}}",
+    units="~",
+    units_long="None",
+    description="Elongation at a toroidal cross-section",
+    dim=1,
+    params=[],
+    transforms={"grid": []},
+    profiles=[],
+    coordinates="z",
+    data=["g_tt", "A"],
+    parameterization=["desc.geometry.surface.FourierRZToroidalSurface"],
+)
+def _a_major_over_a_minor_ToroidalSurface(params, transforms, profiles, data, **kwargs):
```

Comment:
It looks like we could easily combine this with the existing `_a_major_over_a_minor` function to reduce the duplicated code.

Reply:
I thought we couldn't bc the original implementation was only correct for quadrature grids?